src/rpc/handlers/get-providers.js

```diff
@@ -33,7 +33,7 @@ module.exports = (dht) => {
     const dsKey = utils.bufferToKey(cid.buffer)
 
     parallel([
```

Comment:
Sure, I can migrate the internals of this. I was originally trying to avoid touching too much other than the datastore, but I can go ahead and refactor this.

Reply:
I created a new `getProvidersAsync` method that uses all the existing async methods, so no promisify is needed. `getProviders` now calls that with promise-to-callback support, so when we refactor internally we can just do a full swap out of those.